### PR TITLE
test: cover I/O shutdown waker cycles

### PR DESCRIPTION
**Pull request by tomatotomata**
## Motivation

Issue #3481 describes a shutdown cycle where a custom task waker retains the I/O registration that stores it. The shutdown path already clears stored wakers, but this edge case had no regression test.

## Solution

Add an integration test that polls a `TcpListener` with a custom `ArcWake` retaining the same listener. The test verifies that the waker holder remains alive before runtime shutdown and is released when shutdown clears the registration waker.

## Testing

- `cargo fmt --all -- --check`
- `cargo test -p tokio --features full --test io_driver_drop`

### CODE REVIEW

File: tokio/tests/io_driver_drop.rs [hunk 1]
```diff
@@ -5,6 +5,13 @@ use tokio::net::TcpListener;
 use tokio::runtime;
 use tokio_test::{assert_err, assert_pending, assert_ready, task};
 
+use futures::task::{waker_ref, ArcWake};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
```
Comment (Darksonn) [inline tokio/tests/io_driver_drop.rs]: Tokio's usual style is
```suggestion
use std::sync::atomic::{AtomicBool, Ordering};
use std::sync::Arc;
```

File: tokio/tests/io_driver_drop.rs [hunk 2]
```diff
@@ -52,6 +59,55 @@ fn drop_wakes() {
...
+    dropped: Arc<AtomicBool>,
+}
+
+impl ArcWake for RegistrationWaker {
+    fn wake_by_ref(_: &Arc<Self>) {}
+}
+
+impl Drop for RegistrationWaker {
+    fn drop(&mut self) {
+        self.dropped.store(true, Ordering::SeqCst);
```
Comment (Darksonn) [inline tokio/tests/io_driver_drop.rs]: Orderings can be Relaxed

**Review by Darksonn — COMMENTED** [review]
Thanks!

File: tokio/tests/io_driver_drop.rs (hunk 1 above)
Comment (tomatotomata) [reply tokio/tests/io_driver_drop.rs]: Applied the repository-style import split in `tokio/tests/io_driver_drop.rs`.

File: tokio/tests/io_driver_drop.rs (hunk 2 above)
Comment (tomatotomata) [reply tokio/tests/io_driver_drop.rs]: Changed the flag store/load operations to `Ordering::Relaxed`. The flag is only read after the current-thread runtime is dropped, so these operations do not provide cross-thread synchronization. The focused test still passes.

**Review by Darksonn — APPROVED** [review]
Thanks.
